examples/SimilarityScoring.ipynb

```diff
@@ -471,41 +324,89 @@
     }
    ],
    "source": [
+    "# display images and saliency map in a nice format\n",
+    "def show_sal_map(ref_image, query_image, sal_map):\n",
+    "    plt.figure(figsize=(12, 6))\n",
+    "    plt.subplot(2, 3, 1)\n",
+    "    plt.imshow(ref_image)\n",
+    "    plt.axis('off')\n",
+    "    plt.title('Reference Image')\n",
+    "    \n",
+    "    plt.subplot(2, 3, 2)\n",
+    "    plt.imshow(query_image)\n",
+    "    plt.axis('off')\n",
+    "    plt.title('Query Image')\n",
+    "\n",
+    "    # Some magic numbers here to get colorbar to be roughly the same height\n",
+    "    # as the plotted image.\n",
+    "    colorbar_kwargs = {\n",
+    "        \"fraction\": 0.046*(query_image.shape[1]/query_image.shape[0]),\n",
+    "        \"pad\": 0.04,\n",
+    "    }\n",
+    "\n",
+    "    print(f\"Saliency map range: [{sal_map.min():.3g}, {sal_map.max():.3g}]\")\n",
+    "\n",
+    "    # Positive half saliency\n",
+    "    plt.subplot(2, 3, 3)\n",
+    "    plt.imshow(query_image, alpha=0.7)\n",
+    "    plt.imshow(\n",
+    "        np.clip(sal_map, 0, 1),\n",
+    "        cmap='jet', alpha=0.3\n",
+    "    )\n",
+    "    plt.clim(0, 1)\n",
+    "    plt.colorbar(**colorbar_kwargs)\n",
+    "    plt.title(f\"Pos Saliency\")\n",
+    "    plt.axis('off')\n",
+    "\n",
+    "    # Negative half saliency\n",
+    "    plt.subplot(2, 3, 6)\n",
+    "    plt.imshow(query_image, alpha=0.7)\n",
+    "    plt.imshow(\n",
+    "        np.clip(sal_map, -1, 0),\n",
+    "        cmap='jet_r', alpha=0.3\n",
+    "    )\n",
+    "    plt.clim(-1, 0)\n",
+    "    plt.colorbar(**colorbar_kwargs)\n",
+    "    plt.title(f\"Neg Saliency\")\n",
+    "    plt.axis('off')\n",
+    "    plt.show()\n",
+    "    plt.close()\n",
+    "\n",
+    "    \n",
     "# Computing saliency across all combinations of the two test images as query \n",
     "# and reference images for Sliding Window-based perturbation\n",
-    "sal_maps = app(\n",
-    "    test_image1_path,\n",
-    "    test_image2_path,\n",
-    "    simbbox,\n",
-    "    slid_algo,\n",
-    "    similarity_alg,\n",
-    "    fill=simbbox.blackbox_fill,\n",
-    "    vis_mask_examples=True,\n",
+    "\n",
+    "# a -> b\n",
+    "sal_map = sal_generator(\n",
+    "    ref_image=test_image1,\n",
+    "    query_image=test_image2,\n",
+    "    blackbox=desc_gen\n",
+    ")\n",
+    "show_sal_map(test_image1, test_image2, sal_map)\n",
+    "\n",
+    "# b -> a\n",
+    "sal_map = sal_generator(\n",
+    "    ref_image=test_image2,\n",
+    "    query_image=test_image1,\n",
+    "    blackbox=desc_gen\n",
     ")\n",
-    "sal_maps = app(\n",
-    "    test_image2_path, # The change\n",
-    "    test_image1_path, # The change\n",
-    "    simbbox,\n",
-    "    slid_algo,\n",
-    "    similarity_alg,\n",
-    "    fill=simbbox.blackbox_fill,\n",
+    "show_sal_map(test_image2, test_image1, sal_map)\n",
+    "\n",
+    "# a -> a\n",
+    "sal_map = sal_generator(\n",
```

Comment:
The saliency map ranges here for the last two image similarity saliency pairs are slightly different than in master, e.g. [0.0397, 1] vs. [0.0607, 1] and [0.044, 1] vs. [0.079, 1]. Do we know why this is? The shape of the saliency maps look generally consistent though.

Reply:
I believe this is due to the updated sliding window implementation.